client/app/domain/zone/domain-zone-dns.controller.js

```diff
@@ -107,8 +107,9 @@ angular.module("App").controller(
                     this.defaultsDns = _.get(defaults, "paginatedZone.records.results", []).filter((data) => data.subDomain === "" && data.subDomainToDisplay === "").map((value) => value.targetToDisplay.slice(0, -1)).sort();
                     this.activatedDns = _.get(activated, "dns", []).filter((dns) => dns.isUsed).map((value) => value.host).sort();
 
-                    if (!_.isEmpty(this.defaultsDns) && !_.isEqual(this.defaultsDns, this.activatedDns)) {
-                        this.useDefaultsDns = false;
+                    // check for active DNS different than default ones
+                    if (!_.isEmpty(this.activatedDns) && !_.isEmpty(_.xor(this.defaultsDns, this.activatedDns))) {
```

Comment:
Maybe you could extract the condition to make it easier to understand:
`
const someDNSAreActived = !_.isEmpty(this.activatedDns);
const allActivatedDNSAreDefaultDNS = _.isEmpty(_.xor(this.defaultsDns, this.activatedDns)

if (someDNSAreActived && !allActivatedDNSAreDefaultDNS)
`